### DISCUSSION TITLE
Considering transitioning from C99 to C11

### DISCUSSION THREAD

**Discussion by jphickey** [nasa/cFS#748] [Ideas]
Looking for community feedback on this possibility -- ISO9899:2011 (C11) standardizes some of the features that OSAL and CFE/CFS rely on.  In particular:

- Standardized atomic types and atomic ops
- Aligned memory allocation and alignment checks
- Static asserts

Will a transition from C99 to C11 cause any issues for users?  Importantly, this would require all users of CFE to have a C11-compliant toolchain.  For GCC, a version of at least 4.9 would be needed.  As this version is nearing 10 years old, it is expected that most embedded platforms would have caught up by now.   But please share your thoughts if this will cause an issue for any projects.

**Comment by skliper** [comment]
I'm not using or aware of anyone using GCC older than 4.9 w/ modern cFS (or at all really for any current development).

**Reply by skliper** [reply]
Looks like the old ppc-vxworks toolchain is set to gcc 4.3.3:
https://github.com/nasa/cFE/blob/4577b78eb65d0d0038e0cd1653ea4b6aa20dbfe6/cmake/sample_defs/toolchain-ppc-vxworks6.9.cmake#L8

It's probably due an update to VxWorks 7.

**Reply by jphickey** [reply]
Yep, this was discussed in CCB as part of the trade-off.  We'd essentially be stopping support for both RTEMS 4.11 (which we've already kinda-sorta agreed to, but haven't done yet because its still in the workflow) and VxWorks 6.9.  RTEMS 5 and VxWorks 7.x should be fine though.

**Comment by jwberck** [comment]
I'll just pop in and mention that the SPLICE has been using C11 for years without issue in cFS. All newer projects I'm aware of are using C11 or above.

**Comment by LucienMorey** [comment]
I am under the impression that xiphos processor packages are reasonably common in this space. The suggested images for running these chips in linux offer a compiler version newer than the minimum requirement mentioned. With this in mind, I cant see any issues with transitioning from the perspective of users on this platform.
